### ISSUE TITLE
Is sat suuport saving checkpoint by using fp16 or bf16?

### ISSUE THREAD

**Issue by xxxwuwq** [THUDM/SwissArmyTransformer#160]
Is sat suuport saving checkpoint by using fp16 or bf16?

**Comment by 1049451037** [comment]
Saving checkpoint preserves the original parameter dtype. Do you mean you want to train model with fp32, but save it with fp16? If you train a model with fp16, the model will be save with fp16 by default.

**Comment by xxxwuwq** [comment]
Yes, if support, i can chose what i need, cause that when i using cogvlm to finetune, it only support save checkpoint in fp32, which need 60GB+ storage to save one model,  fp16 maybe enough. offical api seems doesn'y support

**Comment by 1049451037** [comment]
cogvlm finetune saves model in fp16, unless you train it with fp32.

**Comment by 1049451037** [comment]
It is memory consuming to save the model with a different dtype with that you train, because you need a copy of the whole model to complete that.

**Comment by xxxwuwq** [comment]
thanks a lot
